omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,245 @@
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = getCreatableProperties();
+		description.removeProperty("identifierType");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		Object identifierTypeUuid = postBody.get("identifierType");
+		ArrayList<String> errors = new ArrayList<String>();
+		if ((sourceUuid == null) || (StringUtils.isBlank(sourceUuid.toString()))) {
+			errors.add("source");
+		}
+		if ((identifierTypeUuid == null) || (StringUtils.isBlank(identifierTypeUuid.toString()))) {
+			errors.add("identifierType");
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException("The values of the following inputs are missing: " + errors.toString());
+		}
+		IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+		if (identifierSource == null) {
+			errors.add("source");
+		}
+		PatientIdentifierType patientIdentifierType = Context.getPatientService()
+		        .getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+		if (patientIdentifierType == null) {
+			errors.add("IdentifierType");
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException("The values of the following inputs are invalid: " + errors.toString());
+		}
+		Location location = null;
+		if ((locationUuid != null) && (StringUtils.isNotBlank(identifierTypeUuid.toString()))) {
+			location = Context.getLocationService().getLocationByUuid(locationUuid.toString());
+		}
+		if (location == null) {
+			location = Context.getLocationService()
+			        .getLocationByUuid(Context.getLocationService().getDefaultLocation().getUuid());
+		}
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		autoGenerationOption.setSource(identifierSource);
+		autoGenerationOption.setIdentifierType(patientIdentifierType);
+		autoGenerationOption.setManualEntryEnabled(this.parseBoolean(manualEntryEnabled.toString()));
+		autoGenerationOption.setAutomaticGenerationEnabled(this.parseBoolean(automaticallyEnabled.toString()));
+		autoGenerationOption.setLocation(location);
+		return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.FULL);
+		
+	}
+	
+	@Override
+	public SimpleObject update(String uuid, SimpleObject postBody, RequestContext context) throws ResponseException {
+		int id = 0;
+		try {
+			id = Integer.parseInt(uuid);
+		}
+		catch (NumberFormatException e) {
+			throw new ValidationException("The uuid enterd is invalid: " + e.getMessage());
```

Comment:
You could make this more user friendly by stating what makes it invalid.

Reply:
<img width="676" alt="screen shot 2017-10-19 at 12 15 54" src="https://user-images.githubusercontent.com/28532877/31763607-712a07fc-b4c7-11e7-8f58-285d8b1d16ac.png">

Reply:
that is when uuid=adadafq

Reply:
@dkayiwa any suggestions on making it more user friendly

Reply:
What makes it invalid?